Add Mammotion integration documentation

mikey0000 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/143429
- Link to parent pull request in the Brands repository:  https://github.com/home-assistant/brands/pull/9020
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

mikey0000 committed:
start docs

mikey0000 committed:
Merge branch 'next' into mammotion-integration-docs

frenck reviewed (changes requested):
Oh hi there @mikey0000 👋 

Thanks for opening the documentation PR for this integration. However...

* Can you make sure the automated build processes passes?
* Please make sure the pull request title is descriptive.

Thanks! 👍 

../Frenck  

<p>
  <a href="https://www.linkedin.com/in/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/linkedin.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://youtube.com/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/youtube.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://bsky.app/profile/frenck.social"><img src="https://github.com/frenck/frenck/raw/main/images/bluesky.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://fosstodon.org/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/mastodon.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.instagram.com/frenck/"><img src="https://github.com/frenck/frenck/raw/main/images/instagram.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.threads.net/@frenck"><img src="https://github.com/frenck/frenck/raw/main/images/threads.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.facebook.com/frenck.dev/"><img src="https://github.com/frenck/frenck/raw/main/images/facebook.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://x.com/frenck"><img src="https://github.com/frenck/frenck/raw/main/images/x.svg" width="18" height="18"></a>
  &nbsp;&nbsp;
  <a href="https://www.tiktok.com/@frenck.nl"><img src="https://github.com/frenck/frenck/raw/main/images/tiktok.svg" width="18" height="18"></a>
</p>

Blogging my personal ramblings at <a href="https://frenck.dev">frenck.dev</a>

mikey0000 committed:
Merge branch 'next' into mammotion-integration-docs

mikey0000 committed:
update further details

mikey0000 committed:
fix google to Google

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+- Return to dock
+- Status monitoring
+
+{% enddetails %}
```
```suggestion
```

c0ffeeca7 committed:
remove tag in attempt to fix build

mikey0000 committed:
Merge branch 'next' into mammotion-integration-docs

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,64 @@
+Email:
+description: Your Mammotion account email
+Password:
+description: Your Mammotion account password
```
```suggestion
Email:
    description: Your Mammotion account email
Password:
    description: Your Mammotion account password
```

c0ffeeca7 committed:
Update source/_integrations/mammotion.markdown

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,64 @@
+- Status monitoring
+
+
+## Known issues
```
```suggestion
## Known issues

```

c0ffeeca7 committed:
Update source/_integrations/mammotion.markdown

c0ffeeca7 reviewed (changes requested):
Please add a link to the related brands PR

mikey0000 committed:
Merge branch 'next' into mammotion-integration-docs

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+ha_quality_scale: bronze
+---
+
+The Mammotion integration provides BLE and Wi-Fi connectivity with Mammotion products from Home Assistant.
```
```suggestion
The **Mammotion** {% term integration %} provides BLE and Wi-Fi connectivity with Mammotion products from Home Assistant.
```

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+
+## Prerequisites
+
+Mammotion logins only allow one device to be logged in at a time. To use this integration you will need to create a separate account just for Home Assistant and share devices to that account.
```
```suggestion
Mammotion logins only allow one device to be logged in at a time. To use this integration, you need to create a separate account just for Home Assistant and share devices with that account.
```

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+
+To create the separate account:
+
+1. Sign up using the Mammotion App with an email address you plan to use just with Home Assistant, SSO such as Google is not officially supported though Google email addresses do work.
```
```suggestion
1. Sign up using the Mammotion App with an email address you plan to use just with Home Assistant. SSO, such as Google, is not officially supported, though Google email addresses do work.
```

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+To create the separate account:
+
+1. Sign up using the Mammotion App with an email address you plan to use just with Home Assistant, SSO such as Google is not officially supported though Google email addresses do work.
+2. Sign into your main account and share the devices you want to control with the new account.
```
```suggestion
2. Sign in to your main account and share the devices you want to control with the new account.
```

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+
+1. Sign up using the Mammotion App with an email address you plan to use just with Home Assistant, SSO such as Google is not officially supported though Google email addresses do work.
+2. Sign into your main account and share the devices you want to control with the new account.
+3. Sign into the new account and accept the shared devices.
```
```suggestion
3. Sign in to the new account and accept the shared devices.
```

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+
+{% include integrations/config_flow.md %}
+
+Use your new account credentials in the the follw
```
```suggestion
Use your new account credentials during setup:
```

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+
+## Lawn mower
+
+The integration will create a lawn mower entity to control the mower. This entity can:
```
```suggestion
The integration creates a lawn mower entity to control the mower. This entity can:
```

c0ffeeca7 committed:
tiny tweaks

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+
+## Known issues
+
+Because Home Assistant now talks to your device, this can sometimes cause issues with making changes using the Mammotion App, though the integration will stop communicating when it sees the device is in setup states, the device will continue to work as expected even if you see a delay visually on your phone when setting up zones, areas or channels. 
```
```suggestion
Because Home Assistant now talks to your device, this can sometimes cause issues with making changes using the Mammotion App. The integration stops communicating when it sees the device is in setup state. However, the device continues to work in the background, even if you notice a delay on your phone when setting up zones, areas, or channels. 
```
Not sure if I fully understood this section. Tried to rephrase it a bit. Please feel free to change it if the edited version is incorrect.

mikey0000 replied:
Thank you, there are a lot of device states E.g create boundary, create zone, create boundary, lock, mowing etc. I'll review this shortly

mikey0000 replied:
I've updated it and tried to make it clearer, please take a look and let me know if it makes better sense

mikey0000 committed:
Improve clarity of known issues in Mammotion integration

Clarified known issues regarding the Mammotion App and Home Assistant integration, emphasizing the behavior during map data modifications.

mikey0000 committed:
Merge branch 'next' into mammotion-integration-docs

klaasnicolaas commented:
> It's already merged, was over a year ago

For a core integration, the asset files would still need to be copied to the other folder (`core_integrations`).

./Klaas

mikey0000 commented:
> > It's already merged, was over a year ago
> 
> For a core integration, the asset files would still need to be copied to the other folder (`core_integrations`).
> 
> ./Klaas

Opened new PR with the images copied over to core. Sorry for my mistake.

c0ffeeca7 commented:
> > It's already merged, was over a year ago
> 
> For a core integration, the asset files would still need to be copied to the other folder (`core_integrations`).
> 
> ./Klaas

Thank you for the catch! :heart:

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+ha_quality_scale: bronze
+---
+
+The **Mammotion** {% term integration %} provides BLE and Wi-Fi connectivity with Mammotion products from Home Assistant.
```
could you add a link to their product website?

c0ffeeca7 commented on source/_integrations/mammotion.markdown:
```diff
@@ -0,0 +1,65 @@
+description: Instructions on how to integrate Mammotion robotic lawn mowers with Home Assistant.
+ha_category:
+  - Lawn Mower
+ha_release: 2026.01
```
```suggestion
ha_release: 2026.2
```

c0ffeeca7 reviewed (changes requested):
tiny tweaks

mikey0000 committed:
Update source/_integrations/mammotion.markdown

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

mikey0000 committed:
add mammotion product url

c0ffeeca7 reviewed (approved):
Thank you, @mikey0000 :+1:

mikey0000 committed:
Merge branch 'next' into mammotion-integration-docs
